src/app/components/task-list/task-list.component.ts

```diff
@@ -34,6 +36,10 @@ export class TaskListComponent implements OnInit, OnDestroy {
   }
 
   removeTask(taskID: string) {
-    this.taskService.removeTaskByID(taskID);
+    this.deleteSubscription = this.taskService.removeTaskByID(taskID).subscribe(() => {
+      this.router.navigateByUrl("/").then(() => {
```

Comment:
You are already on this page though? Why do you want to redirect to the same page? Instead what you could do here, since you remove a task from task-list (and remain on the same page), you can return a list of new tasks and update the tasks subject

Reply:
essentially here you can call the this.taskService.setTasks() method where you pass the new list of tasks you receive as a response from the deleteTask method